### PR TITLE
Issue #630 - Remap element for BrickLink export

### PR DESCRIPTION
**Pull request by hfiguiere**
This is my first prototype at fixing #630.

I'm thinking of changing the following:

- [x] move the code out of project.cpp
- [x] make the map list an external resource: I'm thinking a text file, one key - value per line and `!` for comments

Let me know if you have a different suggestion.

### CODE REVIEW

**Comment by trevorsandy** [comment]
> make the map list an external resource: I'm thinking a text file, one key - value per line and ! for comments

This is how I handle my external mapping lists (BrickLink, LEGO, ReBrickable etc...). I create a static list, which you have done, but on load, I first check if there is an external file and load that instead if it exists. Additionally, Users can choose to edit the list via a menu action at which point I export the static list to disc file, again if it does not exist. With this framework, lists can be maintained by the user. See https://github.com/trevorsandy/lpub3d/blob/master/mainApp/annotations.cpp 

Cheers,

**Comment by leozide** [comment]
When moving to an external file, I think an existing format like json or xml is better. The lxf code is a good example.
Also, would be good to move the color table to the external file.

**Comment by rsbx** [comment]
Something lightweight like Apple's plist would be preferred since it's 
very easy to parse without libraries.

File: common/lc_bricklink.cpp [hunk 1]
```diff
@@ -3,134 +3,15 @@
...
-		{"93888", "3007"}, //  Unoffical LDraw part
-		{"95820", "30237b"},
-		{"973p101", "973ps1"},
-		{"975", "982"},
-		{"976", "981"},
-		{"977", "983"},
-		{"u9026", "3062old"},
-	};
-
+	std::map<std::string, const char*> Remapping;
```
Comment (hfiguiere) [inline common/lc_bricklink.cpp:14]: This should be `std::map<std::string, std::string>`

Comment (hfiguiere) [reply common/lc_bricklink.cpp]: fixed now.

**Comment by hfiguiere** [comment]
> > make the map list an external resource: I'm thinking a text file, one key - value per line and ! for comments
> 
> This is how I handle my external mapping lists (BrickLink, LEGO, ReBrickable etc...). I create a static list, which you have done, but on load, I first check if there is an external file and load that instead if it exists. Additionally, Users can choose to edit the list via a menu action at which point I export the static list to disc file, again if it does not exist. With this framework, lists can be maintained by the user. See https://github.com/trevorsandy/lpub3d/blob/master/mainApp/annotations.cpp

Shall the mapping be shared?

**Comment by hfiguiere** [comment]
> When moving to an external file, I think an existing format like json or xml is better. The lxf code is a good example.

But this data is much simpler.

> Also, would be good to move the color table to the external file.

In the mean time, I moved the colour table to `lc_bricklink.cpp` because it's only used there.

**Comment by hfiguiere** [comment]
Updated and rebased.

**Comment by leozide** [comment]
Is this final? It's marked as a draft.

**Comment by hfiguiere** [comment]
I think it's in a mergeable state. The result is better than before as we can upload more of the parts inventories to BrickLink...

Now a few limitations:
- hose are a problem
- other assemblies might be a problem
- there is a probably a longer list of parts that don't work

Possible subsequent developments could be an ability for the user to override mappings.

As for the data format, I really don't feel like it should be more complicated, but if you feel something else would be better, then I can work something out.

**Comment by leozide** [comment]
Looks good, thanks for the feature. I think I'll update the file to include the color mappings.

**Comment by hfiguiere** [comment]
do colors really change that often?

**Comment by leozide** [comment]
Not really, but it's nicer than having a big table in code.

**Pull request merged**